lib/lib-search/src/main/java/org/sagebionetworks/search/SearchDaoImpl.java

```diff
@@ -130,18 +124,26 @@ public void createOrUpdateSearchDocument(List<Document> batch) throws ClientProt
 			for(Document document: documents){
 				prepareDocument(document);
 				serializedDocument = EntityFactory.createJSONStringForEntity(document);
-				// AwesomeSearch pukes on control characters. Some descriptions have
-				// control characters in them for some reason, in any case, just get rid
-				// of all control characters in the search document
-				String cleanedDocument = serializedDocument.replaceAll("\\p{Cc}", "");

```

Comment:
I suggest cleaning up the garbage before putting it in the index.  I think this would be done in  SearchDocumentDriverImpl.getAllWikiPageText(String nodeId), but you should probably double-check with John that this is the right place to do it.